sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupClient.java

```diff
@@ -0,0 +1,86 @@
+        this.asyncClient = asyncClient;
+    }
+
+    /**
+     * Get the vault endpoint URL.
+     *
+     * @return The vault endpoint URL.
+     */
+    public String getVaultUrl() {
+        return asyncClient.getVaultUrl();
+    }
+
+    /**
+     * Initiates a full backup of the Key Vault.
+     *
+     * @param blobStorageUrl The URL for the Blob Storage resource where the backup will be located.
+     * @param sasToken       A Shared Access Signature (SAS) token to authorize access to the blob.
+     * @return A {@link SyncPoller} polling on the {@link FullBackupOperation backup operation} status.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code blobStorageUrl} or {@code sasToken} are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public SyncPoller<FullBackupOperation, Void> startBackup(String blobStorageUrl, String sasToken) {
+        return asyncClient.startBackup(blobStorageUrl, sasToken).getSyncPoller();
+    }
+
+    /**
+     * Initiates a full restore of the Key Vault.
+     *
+     * @param blobStorageUrl The URL for the Blob Storage resource where the backup is located.
+     * @param sasToken       A Shared Access Signature (SAS) token to authorize access to the blob.
+     * @param folderName     The name of the folder containing the backup data to restore.
+     * @return A {@link SyncPoller} polling on the {@link RestoreOperation backup operation} status.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code blobStorageUrl} or {@code sasToken} are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public SyncPoller<RestoreOperation, Void> startRestore(String blobStorageUrl, String sasToken, String folderName) {
+        return asyncClient.startRestore(blobStorageUrl, sasToken, folderName).getSyncPoller();
+    }
+
+    /**
+     * Restores all versions of a given key using the supplied SAS token pointing to a previously stored Azure Blob storage backup folder.
+     *
+     * @param keyName        The name of the key to be restored.
+     * @param blobStorageUrl The URL for the Blob Storage resource where the backup is located.
+     * @param sasToken       A Shared Access Signature (SAS) token to authorize access to the blob.
+     * @param folderName     The name of the folder containing the backup data to restore.
+     * @return A {@link PollerFlux} polling on the {@link RestoreOperation backup operation} status.
+     * @throws KeyVaultErrorException if the operation is unsuccessful.
+     * @throws NullPointerException   if the {@code blobStorageUrl} or {@code sasToken} are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public SyncPoller<SelectiveKeyRestoreOperation, Void> startSelectiveRestore(String keyName, String blobStorageUrl, String sasToken, String folderName) {
```

Comment:
@christothes I don't see this in .NET. Should it be?

Reply:
I added this because I remember us talking about having a selective restore operation in one of our team meetings. Not sure if that's still the plan.

Reply:
I omitted this from the .NET client until the swagger update goes through to generalize the `FullRestoreStatus` method to `RestoreStatus` as it is shared across selective and full restore methods. Initially the swagger only called out the LRO endpoints for full, but I verified with the service team that both should leverage the same status endpoint. The PR is here https://github.com/Azure/azure-rest-api-specs/pull/9842